src/persist/src/indexed/trace.rs

```diff
@@ -127,6 +126,27 @@ impl<K: Data, V: Data> BlobTrace<K, V> {
         }
         Ok(TraceSnapshot { ts_upper, updates })
     }
+
+    /// Update the compaction frontier to `since`.
+    pub fn allow_compaction(&mut self, since: Antichain<u64>) -> Result<(), Error> {
+        if PartialOrder::less_equal(&self.ts_upper(), &since) {
+            return Err(Error::from(format!(
+                "invalid compaction at or in advance of trace upper {:?}: {:?}",
+                self.ts_upper(),
+                since,
+            )));
+        }
+
+        if PartialOrder::less_than(&since, &self.since) {
+            return Err(Error::from(format!(
```

Comment:
Should this be an error or a no-op? I'm divided but a very slight lean toward no-op. What was your thinking here?

Reply:
My thought was that it would be nice if `allow_compaction`'s semantics matched `seal`'s semantics as much as possible (and perhaps that means we should go all the way and make this `less_equal` here rather than `less_than`)

Reply:
I made allow_compaction's requirements as strict as seal's